### ISSUE TITLE
Application lifecycle properties with default values when privacy mode set in OPTIN after EXEMPT

### ISSUE THREAD

**Issue by axel-lefaux** [at-internet/piano-analytics-android#9]
HI !

Something going wrong with application lifecycle properties.

I have to use a consent SDK which is enable or not your analytics SDK.
The user can change the privacy mode EXEMPT to OPTIN and OPTIN to EXEMPT.

And the issue is when the mode change from EXEMPT to OPTIN, all lifecycle properties have default values.
For exemple `app_fsd` is equals to `19700101`

You can reproduce it with little changes in MainActivity is your sample.

```
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.recyclerview.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = MainAdapter(animals, this@MainActivity::onItemClick)
        }

        lifecycleScope.launch {
            delay(10_000)
            PianoAnalytics.getInstance().privacyModesStorage.currentMode = PrivacyMode.EXEMPT
        }
    }

    override fun onResume() {
        super.onResume()
        PianoAnalytics.getInstance().privacyModesStorage.currentMode = PrivacyMode.OPTIN
        
        PianoAnalytics.getInstance().sendEvents(
            Event.Builder(Event.PAGE_DISPLAY)
                .properties(
                    Property(PropertyName.PAGE_FULL_NAME, "main"),
                )
                .build()
        )
    }
```

Start the app, wait for 10 secondes and click on an animal. Then come back to the list and you can see in logs, default values.

**Comment by DeKaN** [comment]
Hi @axel-lefaux,

As I see the issue appears only after changing PrivacyMode to a less strict mode. It persists until app's restart for `app_fsd`, and for other properties until "session expiration" (`configuration.sessionBackgroundDuration` seconds not in foreground mode) or app's restart.

Thanks for finding it.

**Comment by DeKaN** [comment]
Released in version 3.3.3
